### PR TITLE
[DOCS] ES|QL limitations

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Finalizes the ES|QL "Limitations" page in the documentation.

### CODE REVIEW

**Comment by abdonpijpelink** [comment]
@elasticmachine test this please

**Review by luigidellaquila — APPROVED** [review]
LGTM
I just left a small note about `text` fields with synthetic source

File: docs/reference/esql/esql-limitations.asciidoc [hunk 1]
```diff
@@ -5,36 +5,143 @@
...
+By default, an {esql} query returns up to 500 rows. You can increase the number
+of rows up to 10,000 using the <<esql-limit>> command. Queries do not return
+more than 10,000 rows, regardless of the `LIMIT` command's value.
+
+This limit only applies to the number of rows that are retrieved by the query
+and displayed in Discover. Queries and aggregations run on the full data set.
+
+To overcome this limitation:
+
+* Reduce the result set size by modifying the query to only return interesting
```
Comment (astefan) [inline docs/reference/esql/esql-limitations.asciidoc:21]:
```suggestion
* Reduce the result set size by modifying the query to only return relevant
```

File: docs/reference/esql/esql-limitations.asciidoc [hunk 2]
```diff
@@ -5,36 +5,143 @@
...
+** `integer_range`
+** `ip_range`
+** `long_range`
+** `nested`
+** `rank_feature`
+** `rank_features`
+** `search_as_you_type`
+
+Columns with unsupported field types are returned with `null` values, with the
+exception of nested fields. Nested fields are not returned at all.
```
Comment (astefan) [inline docs/reference/esql/esql-limitations.asciidoc:83]: There is another nuance here. A query like `from test | where k8s.pod.network.tx == 1434577921` where `k8s.pod.network.tx` is tsdb `counter` field will return an error: `Cannot use field [k8s.pod.network.tx] with unsupported type [counter]`.

So, in other words, the unsupported data types can be returned and they are null only in those queries where those specific unsupported fields are not explicitly used.

File: docs/reference/esql/esql-limitations.asciidoc [hunk 3]
```diff
@@ -5,36 +5,143 @@
...
+
+[discrete]
+[[esql-limitations-text-fields]]
+=== `text` fields behave like `keyword` fields
+
+While {esql} supports <<text,`text`>> fields, {esql} does not treat these fields
+like the Search API does. {esql} queries do not query or aggregate the
+<<analysis,analyzed string>>. Instead, an {esql} query will try to get a `text`
+field's subfield of type `keyword` and query/aggregate that. If it's not
```
Comment (astefan) [inline docs/reference/esql/esql-limitations.asciidoc:92]: Another nuance here. This is not strictly the type `keyword`, but the keyword family type https://www.elastic.co/guide/en/elasticsearch/reference/current/keyword.html#keyword.

File: docs/reference/esql/esql-limitations.asciidoc [hunk 4]
```diff
@@ -5,36 +5,143 @@
...
+
+Note that {esql}'s retrieval of `keyword` subfields may have unexpected
+consequences. An {esql} query on a `text` field is case-sensitive. Furthermore,
+a subfield may have been mapped with a <<normalizer,normalizer>>, which can
+transform the original string. Or it may have been mapped with <<ignore-above>>,
+which can truncate the string. None of these mapping operations are applied to
+an {esql} query, which may lead to false positive or negatives. 
+
+To avoid these issues, a best practice is to be explicit about the field that
+you query, and query `keyword` fields instead of `text` fields.
```
Comment (astefan) [inline docs/reference/esql/esql-limitations.asciidoc:105]:
```suggestion
you query, and query `keyword` sub-fields instead of `text` fields.
```

**Review by astefan — APPROVED** [review]
Looking great. LGTM!

File: docs/reference/esql/esql-limitations.asciidoc (hunk 2 above)
Comment (abdonpijpelink) [reply docs/reference/esql/esql-limitations.asciidoc]: Thanks! I have reworded it as: _Querying a column with an unsupported type returns an error. If a column with an unsupported type is not explicitly used in a query, it is returned with `null`values, with the exception of nested fields. Nested fields are not returned at all._

**Pull request merged**